gradle/dependencies.gradle

```diff
@@ -70,8 +70,9 @@ else
 // The default strategy (i.e. what `mockito-core` uses) changed to `inline` in 5.x because it works better with newer
 // Java versions.
 // We always use the `inline` strategy.
+// For Scala 2.12 we have to use the mockito-inline library because otherwise we cannot return mocks of final classes.
```

Comment:
This comment is not accurate, `mockito-core` from version `5` does support that. The issue is that Scala 2.12 uses mockito 4.

Reply:
Yes, you are absolutely correct. Unfortunately, even with the additional check I still cannot quite make all the tests to pass, so I will continue battling this. But thank you for the review!

Reply:
Yes, we don't run Scala 2.12 with Java 11+ so it would not impact the PR builds.